credentials-migrate.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/sh
+
+if [ $# -lt 1 ]
+then
+  echo "Application ID must be supplied. Usage: ./credentials-migrate.sh <app-id>"
+  exit 1
+fi
+APPLICATION=$1
+
+which jq > /dev/null && which drg > /dev/null
+if [ $? -ne 0 ]
+then
+  echo "jq and drg not found. These tools are mandatory to run this script"
+  exit 1
+fi
+
+# test for a valid drg config
+CONFIG=$(drg config show --active -o json | jq .name)
+if [ $? -ne 0 ]
+then
+  echo "drg is not logged in. Set up drg before continuing with: drg login <url>"
+  exit 1
+fi
+
+echo "Using drg config $CONFIG"
+
+# get a list of device | keep only column 1 | remove header
+DEVICES_ID=$(drg get devices --app "$APPLICATION" | awk '{print $1}' | tail -n +2)
+
+for dev in $DEVICES_ID
+do
+  DATA=$(drg get device "$dev" -o json --app "$APPLICATION")
+  hascreds=$(echo "$DATA" | jq '.spec | has("credentials")')
+
+  if [ "$hascreds" = "true" ]
+  then
+    DATA=$(echo "$DATA" | jq '.spec.authentication = .spec.credentials | del(.spec.credentials)')
```

Comment:
Wouldn't this override any existing `.spec.authentication` section? Shouldn't we rather "merge" this?

Reply:
so we have both ?  Yes we could, then we need to document / explain it thoroughly because it might be confusing :)

Reply:
@lulf I think that's correct, we have both, right?

Reply:
We should merge them, yes. We have both:

1) .credentials is deprecated, but still supported reading until 0.12.0
2) .authentication might be used before running the migration script, in which case you'll have both of them